e2e/docs/docs_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 The envd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docs
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/tensorchord/envd/pkg/app"
+	"github.com/tensorchord/envd/pkg/home"
+)
+
+var _ = FDescribe("check examples in documentation", func() {
+	buildContext := "doctest"
+	// env := "up-test"
+	baseArgs := []string{
+		"envd.test", "--debug",
+	}
+
+	BeforeAll(func() {
+		Expect(home.Initialize()).NotTo(HaveOccurred())
+		envdApp := app.New()
+		err := envdApp.Run(append(baseArgs, "bootstrap"))
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("can list envd envs", func() {
+		envdApp := app.New()
+		err := envdApp.Run([]string{"envd.test", "--debug", "envs", "list"})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Context("check envd envs details", func() {
+		buildContext := "testdata/minimal"
+		args := append(baseArgs, []string{
+			"up", "--path", buildContext, "--detach", "--force",
+		}...)
+		envdApp := app.New()
+		err := envdApp.Run(args)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = envdApp.Run([]string{"envd.test", "--debug", "envs", "describe"})
+		Expect(err).NotTo(HaveOccurred())
+
+		destroyArgs := append(baseArgs, []string{
+			"destroy", "--path", buildContext,
+		}...)
+		err = envdApp.Run(destroyArgs)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	up_tests := []string{"testdata/minimal", "testdata/getting_started", "testdata/jupyter", "testsdata/complex"}
```

Comment:
Should we move everything to `examples` and then test things in that folder directly? Otherwise, we need to maintain the copies in this folder.

Reply:
It's for the examples on the documentation site. Therefore no duplication here I think

Reply:
I wonder if we can insert these `envd` files into the doc instead of copying them here.

Reply:
It would be great but may be hard to implement. I think copy/paste works in the current design.

Reply:
https://vitepress.vuejs.org/guide/markdown#import-code-snippets
Vitepress support this, but bit hard since they're not in the same repo

Reply:
Will open a new issue to track this.